### PR TITLE
Declare and install generic algorithms for projective and injective dimension

### PR DESCRIPTION
**Pull request by sebastianpos**
My introduction of the type `non_neg_integer_or_infinity` leads to deletions of precompiled code, which I think is a bad sign.

@zickgraf Did I forget something in the introduction of that type, or do you need to adjust the compiler?

### CODE REVIEW

File: CAP/gap/CategoryObjects.gd [hunk 1]
```diff
@@ -416,3 +416,25 @@ DeclareOperation( "SimplifyObject_IsoFromInputObject",
...
+###################################
+##
+#! @Section Dimensions
+##
+###################################
+
+#! @Description
+#! The argument is an object $A$.
+#! The output is a the projective dimension of $A$.
+#! @Returns a non negative integer or infinity
```
Comment (mohamed-barakat) [inline CAP/gap/CategoryObjects.gd:429]:
```suggestion
#! @Returns a nonnegative integer or infinity
```

Comment (mohamed-barakat) [inline CAP/gap/CategoryObjects.gd:429]:
```suggestion
#! @Returns a nonnegative integer or infinity
```

File: CAP/gap/CategoryObjects.gd [hunk 2]
```diff
@@ -416,3 +416,25 @@ DeclareOperation( "SimplifyObject_IsoFromInputObject",
...
+#! @Arguments A
+DeclareAttribute( "ProjectiveDimension",
+                  IsCapCategoryObject );
+
+#! @Description
+#! The argument is an object $A$.
+#! The output is a the injective dimension of $A$.
+#! @Returns a non negative integer or infinity
```
Comment (mohamed-barakat) [inline CAP/gap/CategoryObjects.gd:437]:
```suggestion
#! @Returns a nonnegative integer or infinity
```

File: FreydCategoriesForCAP/gap/precompiled_categories/CategoryOfColumnsAsOppositeOfCategoryOfRowsOfFieldPrecompiled.gi [hunk 3]
```diff
@@ -708,172 +708,6 @@ end
         
     , 100 );
     
-    ##
```
Comment (mohamed-barakat) [inline FreydCategoriesForCAP/gap/precompiled_categories/CategoryOfColumnsAsOppositeOfCategoryOfRowsOfFieldPrecompiled.gi:711]: Why did this code disappear?

**Comment by zickgraf** [comment]
The code disappears because `CategoryOfColumns` uses `WrapperCategory` which uses the internal `CategoryConstructor` which should also be adapted here

https://github.com/homalg-project/CAP_project/blob/95ed52baa29546b260ecb8609f2a1200c967161f/CAP/gap/CategoryConstructor.gi#L247

and here

https://github.com/homalg-project/CAP_project/blob/95ed52baa29546b260ecb8609f2a1200c967161f/CAP/gap/CategoryConstructor.gi#L329

That's also the reason why the tests are failing. Sorry, I didn't think of this when we talked about it.

**Comment by zickgraf** [comment]
Addendum: That `CategoryConstructor` does not fail with an error is a deliberate choice. This way, adding new filter types does not cause errors. The downside is that renaming filter types as in this PR silently skips operations which were previously installed. However, since this happens very rarely I see no need for action.

**Comment by sebastianpos** [comment]
Thanks for spotting the typo and thanks for the explanation of the missing code. I replaced `IsCyclotomic` in CategoryConstructor by the new type (which I think is appropriate since `IsCyclotomic` was not a proper solution anyway).

**Comment by zickgraf** [comment]
> I replaced `IsCyclotomic` in CategoryConstructor by the new type (which I think is appropriate since `IsCyclotomic` was not a proper solution anyway).

Yes!

**Pull request merged**